lib/pundit/matchers.rb

```diff
@@ -213,6 +213,26 @@ def configuration
       @forbidden_actions.empty?
     end
 
+    match_when_negated do |policy|
+      ::Kernel.warn 'Using expect { }.not_to permit_actions could produce confusing \
+        results. Please use `.to forbid_actions` instead. To clarify, \
+        `.not_to permit_actions` will look at all of the actions and checks \
+        if ANY actions fail, not if all actions fail. Therefore, you could \
+        result in something like this: \
+
+        it { is_expected.to permit_actions([:new, :create, :edit]) } \
+        it { is_expected.not_to permit_actions([:edit, :destroy]) } \
+
+        In this case, edit would be true and destroy would be false, but both \
+        tests would pass.'
+
+      return true if actions.count < 1
+      @forbidden_actions = actions.reject do |action|
+        policy.public_send("#{action}?")
+      end
+      !@forbidden_actions.empty?
```

Comment:
So my first iteration actually inverted this to `@allowed_actions`, but ultimately that is the `forbid_actions` behavior, so I just decided it to be an explicit `!` of the `permit_actions` behavior.